services/jupyter/src/notebooks/rubrix/rubrix-active-learning-loop.ipynb

```diff
@@ -0,0 +1,701 @@
+       "        features: ['text', 'label'],\n",
+       "        num_rows: 24000\n",
+       "    })\n",
+       "    test: Dataset({\n",
+       "        features: ['text', 'label'],\n",
+       "        num_rows: 7600\n",
+       "    })\n",
+       "})"
+      ]
+     },
+     "execution_count": 60,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "labeled_dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 61,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "DatasetDict({\n",
+       "    unlabeled: Dataset({\n",
+       "        features: ['text'],\n",
+       "        num_rows: 58660\n",
+       "    })\n",
+       "})"
+      ]
+     },
+     "execution_count": 61,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "unlabeled_dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Training data\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 118,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# text\n",
+    "weak_train_text = [row[\"text\"] for row in labeled_dataset[\"train\"]][:100]\n",
+    "unlabeled_train_text = [row[\"text\"] for row in unlabeled_dataset[\"unlabeled\"]][:200]\n",
+    "# labels\n",
+    "weak_train_labels = [row[\"label\"] for row in labeled_dataset[\"train\"]][:100]\n",
+    "unlabeled_no_labels = [LABEL_UNLABELED for _ in unlabeled_train_text][:200]\n",
+    "# dataset\n",
+    "train_text = np.array(weak_train_text + unlabeled_train_text)\n",
+    "train_labels = np.array(weak_train_labels + unlabeled_no_labels)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Preprocessing the dataset\n",
+    "\n",
+    "Wrap the dataset in a specific data class by small-text -> `TransformersDataset` since we are gonna use `distilbert-base-uncased`\n",
+    "\n",
+    "`TransformersDataset` should contain the tokenized input text.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 119,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/src/.venv/lib/python3.10/site-packages/small_text/utils/annotations.py:67: ExperimentalWarning: The function from_arrays is experimental and maybe subject to change soon.\n",
+      "  warnings.warn(f'The {subject} {func_or_class.__name__} is experimental '\n"
+     ]
+    }
+   ],
+   "source": [
+    "# create dataset\n",
+    "dataset = TransformersDataset.from_arrays(\n",
+    "    train_text,\n",
+    "    train_labels,\n",
+    "    tokenizer=TOKENIZER,\n",
+    "    target_labels=[0, 1, 2, 3]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 64,
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "IndexError",
+     "evalue": "list index out of range",
+     "output_type": "error",
```

Comment:
There is an error in the output 👀

Reply:
Yeah my bad. I created subsets for testing given that initializing the model takes quite a bit of time and I forgot to revert it back:

```
weak_train_text = [row["text"] for row in labeled_dataset["train"]][:100]
unlabeled_train_text = [row["text"] for row in unlabeled_dataset["unlabeled"]][:200]
# labels
weak_train_labels = [row["label"] for row in labeled_dataset["train"]][:100]
unlabeled_no_labels = [LABEL_UNLABELED for _ in unlabeled_train_text][:200]
```
I'll fix this.